scripts/manage-db.py

```diff
@@ -13,22 +14,77 @@
 
 from auslib.db import AUSDatabase
 
+
+def cleanup_releases(trans, nightly_age, dryrun=True):
+    # This and the subsequent queries use "%%%%%" because we end up going
+    # through two levels of Python string formatting. The first is here,
+    # and the second happens at a low level of SQLAlchemy when the transaction
+    # is being executed.
+    query = """
+LEFT JOIN releases_history USING (name)
+LEFT JOIN rules rules_mapping ON (name=rules_mapping.mapping)
+LEFT JOIN rules rules_whitelist ON (name=rules_whitelist.whitelist)
+WHERE name LIKE '%%%%nightly%%%%'
+AND (rules_whitelist.whitelist <> releases.name OR rules_whitelist.whitelist IS NULL)
+AND (rules_mapping.mapping <> releases.name OR rules_mapping.mapping IS NULL)
+AND (timestamp < (1000 * UNIX_TIMESTAMP(NOW() - INTERVAL %s MONTH)) OR change_id is NULL);
```

Comment:
Re: lifetime of unmapped releases. I agree that it's a bit confusing. The original idea was that we would only keep history for nightly releases for 7 or 14 days, but we'd keep the actual release for 6 months. When it was written, nobody realized that we'd actually remove the releases after 7 or 14 days as well, because we were using releases_history to determine age. Reading the query after reading your comment makes me realize just how misleading it is -- a newcomer would rightly come to the conclusion that we keep nightly releases for 6 months, even though that's not the case. Either of your suggestions (just looking for releases without history, or parsing the name) sound reasonable to me, we just need to decide which direction to go.

Re: potential edge cases around restoration...I'm not particularly concerned about them, as any restoration from backup should include history. If we're doing direct db inserts....I think we deserve what we get. As this new job is expected to run every 15-30min, a case like that would get noticed quite quickly as well. Do push on this if you feel it's important though.

How strongly do you feel about keeping actual releases for longer than their history? You identified the not-notified-soon-enough watershed, and I can't think of any other case we'd want them. For the watersheds, we could regenerate them if needed with something like https://github.com/mozilla/balrog/blob/master/scripts/generate-json.py (though admittedly it's a PITA, and we'd have to write new code to do it against S3.

Re: the rules conditions - you're right - the <> part should be unnecessary because of the JOIN. I just tried some SELECTs against prod, and this query returns the same results as the one in this PR:
```SELECT name, rules_mapping.mapping, rules_whitelist.whitelist FROM releases LEFT JOIN releases_history USING (name) LEFT JOIN rules rules_mapping ON (name=rules_mapping.mapping) LEFT JOIN rules rules_whitelist ON (name=rules_whitelist.whitelist) WHERE name LIKE '%%%%nightly%%%%' AND rules_whitelist.whitelist IS NULL AND rules_mapping.mapping IS NULL AND (change_id IS NULL OR timestamp < (1000 * UNIX_TIMESTAMP(NOW() - INTERVAL 6 month)));```


To summarize, I need to:
* Get rid of the unnecessary <> checks.
* Actually save releases for the time specified, or change the query to not look as if it's going to without doing it.